ipfs-cluster-service/state.go

```diff
@@ -44,11 +46,62 @@ func upgrade() error {
 	}
 
 	// Reset with SnapshotSave
-	err = raft.SnapshotSave(consensusCfg, newState, clusterCfg.ID)
+	return raft.SnapshotSave(consensusCfg, newState, clusterCfg.ID)
+}
+
+func export() error {
+	cfg, clusterCfg, _, _, consensusCfg, _, _, _, _ := makeConfigs()
+
+	// Execution lock
+	err := locker.lock()
 	if err != nil {
 		return err
 	}
-	return nil
+	defer locker.tryUnlock()
+
+	err = cfg.LoadJSONFromFile(configPath)
+	if err != nil {
+		return err
+	}
+
+	r, snapExists, err := raft.LastStateRaw(consensusCfg)
+	if err != nil {
+		return err
+	}
+	if !snapExists {
+		return errNoSnapshot
+	}
+
+	stateToExport := mapstate.NewMapState()
+	err = stateToExport.Restore(r)
+	if err != nil {
+		return err
+	}
+	return clusterCfg.ExportState(stateToExport)
+}
+
+func stateImport(r io.Reader) error {
```

Comment:
I think before calling this we need to explain the user that the peer's state is going to be fully replaced.

Also:

* Does import need to run in all peers?
* Probably import is not meant to run in a running cluster, so we need to lock (same for export).

Reply:
Are you thinking an interactive "Do you want to override state [y/n]" prompt would be good here?  

In most cases I think import would need to be run in all peers.  The implementation calls raft.SnapshotSave which if you recall takes the term and index values from the latest snapshot.  If all peers saved at the same place along the log then you would see conflicts if their states differed.  However if a peer was already ahead it could import ok because other peers would need to catch up anyway.  For UX purposes its probably best to say that it should be done on all peers.  

I'll put the warning that all peers need to import the same thing alongside the [y/n] prompt (assuming this is what you are suggesting).

Reply:
Also in case it wasn't clear the way this is set up right now the state, while difficult for a user to reset, is not totally wiped from disk, just rotated to an old folder by SnapshotSave's internal cleanup operation.  Regardless I'm in favor of the prompt.

Reply:
Sorry, just saw this after the new review. Maybe the new snapshot should have the last snapshot's index + 1. In this case, maybe all peers would receive the newest snapshot. I am not 100% certain. Let's leave it like it is.